cleanlab/rank.py

```diff
@@ -152,14 +164,320 @@ def get_normalized_margin_for_each_label(
+            """
+        )
+
+    # Adjust predicted probabilities
+    if adj_pred_probs:
+        pred_probs = subtract_confident_thresholds(labels, pred_probs)
+
+    # Pass keyword arguments for scoring function
+    input = {"labels": labels, "pred_probs": pred_probs}
+
+    # Calculate scores
+    label_quality_scores = scoring_func(**input)
+
+    return label_quality_scores
+
+
+def score_label_quality_ensemble(
+    labels: np.array,
+    pred_probs_list: List[np.array],
+    method: str = "self_confidence",
+    adj_pred_probs: bool = False,
+) -> np.array:
+    """Returns the ensemble label quality scores.
+
+    This requires a list of pred_probs from each model in the ensemble.
+
+    The scoring methods are heuristics to enable users to quickly rank order data to find potential label quality issues.
+
+    Parameters
+    ----------
+    labels : np.array
+      A discrete vector of noisy labels, i.e. some labels may be erroneous.
+      *Format requirements*: for dataset with K classes, labels must be in {0,1,...,K-1}.
+
+    pred_probs_list : List of np.array (shape (N, K))
+      List of pred_probs from each model in the ensemble.
+      P(label=k|x) is a matrix with K model-predicted probabilities.
+      Each row of this matrix corresponds to an example x and contains the model-predicted
+      probabilities that x belongs to each possible class.
+      The columns must be ordered such that these probabilities correspond to class 0,1,2,...
+      `pred_probs` should have been computed using 3 (or higher) fold cross-validation.
+
+    method : {"self_confidence", "normalized_margin", "confidence_weighted_entropy"}, default="self_confidence"
+      Label quality scoring method. Default is "self_confidence".
+
+      .. seealso::
+        :func:`self_confidence`
+        :func:`normalized_margin`
+        :func:`confidence_weighted_entropy`
+
+    adj_pred_probs : bool, default = True
+      Adjust predicted probabilities by subtracting the class confident thresholds and renormalizing.
+      The confident class threshold for a class j is the expected (average) "self-confidence" for class j.
+      See paper "Confident Learning: Estimating Uncertainty in Dataset Labels" by Northcutt et al.
+      https://arxiv.org/abs/1911.00068
+
+    Returns
+    -------
+    label_quality_scores : np.array (float)
+
+    See Also
+    --------
+    self_confidence
+    normalized_margin
+    confidence_weighted_entropy
+    score_label_quality
+
+    """
+
+    # Generate scores for each model's pred_probs
+    scores_list = []
+    accuracy_list = []
+    for pred_probs in pred_probs_list:
+
+        # Calculate scores and accuracy
+        scores = score_label_quality(
+            labels=labels,
+            pred_probs=pred_probs,
+            method=method,
+            adj_pred_probs=adj_pred_probs,
+        )
+        accuracy = (pred_probs.argmax(axis=1) == labels).mean()
```

Comment:
I don't understand why you're computing this. Its in a for-loop but none of the inputs to accuracy change, so its just computing the same thing over and over. So, then what are the weights if all the weights are the same number? I'm confused.

Reply:
The for-loop goes through each `pred_probs` in `pred_probs_list` (one `pred_probs` for each model in the ensemble).

The inputs to accuracy will change in the for-loop because `pred_probs` changes. The weights will change as the accuracy changes.